### PR TITLE
ENH: expose dwithin on the Geometry subclasses

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
xref https://github.com/shapely/shapely/issues/1281

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
Are we OK with also adding `contains_properly`?

File: shapely/geometry/base.py [hunk 1]
```diff
@@ -618,6 +618,14 @@ def within(self, other):
         """Returns True if geometry is within the other, else False"""
         return bool(shapely.within(self, other))
 
+    def dwithin(self, other, distance):
+        """
+        Returns True if geometry is within a given distance from the other, else False.
+
+        Refer to `shapely.dwithin` for full documentation.
```
Comment (brendan-ward) [inline shapely/geometry/base.py:625]: One thing I noticed in the generated docs (unless I'm building them incorrectly) is that the Shapely docs do not add a note about the required version of GEOS, whereas pygeos included a note about this.

Does this also need the `@requires_geos("3.10.0")` annotation here?

**Review by brendan-ward — COMMENTED** [review]
Thanks for working on this!

File: shapely/geometry/base.py (hunk 1 above)
Comment (mwtoews) [reply shapely/geometry/base.py:625]: no annotation is required, a helpful exception is raised: "shapely.errors.UnsupportedGEOSVersionError: 'dwithin' requires at least GEOS 3.10.0"

**Review by mwtoews — APPROVED** [review]
Thanks, local testing looks good! I can't wait to test this out more thoroughly, as I consider [ST_DWithin](https://postgis.net/docs/ST_DWithin.html) one of the best PostGIS functions.

File: shapely/geometry/base.py (hunk 1 above)
Comment (jorisvandenbossche) [reply shapely/geometry/base.py:625]: Ah, that's a good point about the docstrings not including those notes. 
@mwtoews from pygeos, we had an automated injection of a note into the docstring about the required GEOS version, see eg https://pygeos.readthedocs.io/en/stable/constructive.html#pygeos.constructive.build_area vs https://shapely.readthedocs.io/en/latest/constructive.html#shapely.constructive.build_area 

The code for this is still included, so not really sure why this is not working .. (we also have tests for this, and those are passing)

Comment (jorisvandenbossche) [reply shapely/geometry/base.py:625]: Fixing the docstring note in https://github.com/shapely/shapely/pull/1519

**Pull request merged**